src/cascadia/TerminalControl/ThrottledFunc.h

```diff
@@ -81,93 +133,46 @@ class ThrottledFunc : public std::enable_shared_from_this<ThrottledFunc<Args...>
     template<typename F>
     void ModifyPending(F f)
     {
-        std::lock_guard guard{ _lock };
-
-        if (_pendingRunArgs.has_value())
-        {
-            std::apply(f, _pendingRunArgs.value());
-        }
+        _storage.ModifyPending(f);
     }
 
 private:
-    static winrt::fire_and_forget _Fire(winrt::Windows::Foundation::TimeSpan delay, winrt::Windows::UI::Core::CoreDispatcher dispatcher, std::weak_ptr<ThrottledFunc> weakThis)
+    winrt::fire_and_forget _Fire()
     {
-        co_await winrt::resume_after(delay);
-        co_await winrt::resume_foreground(dispatcher);
```

Comment:
so, this is an interesting semantic change! It used to be guaranteed that the throttled function was called on the foreground queue; now that guarantee is only true for Trailing

Reply:
Oh dang I haven't thought of that. Good catch!